### ISSUE TITLE
OBS crashes on Mac M3 Pro when transitioning through scenes. 

### ISSUE THREAD

**Issue by felixolivares** [exeldro/obs-move-transition#241]
I have a constant crash issue on my Mac installation. It is not really consistent but it happens every time I open OBS. 

The steps to reproduce this are not very constant but most of them is when I transition between scenes using the Move transition. I thought the issue was with other plugins I recently installed but between installing and installing all my plugins I noticed that the sad winner is Move Transition plugin. When installed and using it to transition between scenes the crash happens quite regularly. 

I use OBS for live streaming on Facebook. The scenes that I use where the crash happen is both scenes have a video input: One source is the iPhone camera which I use as a main feed and the FaceTime camera of the Mac (which I use as a secondary camera). Not sure if this might matter but I just let you know this in advance. 

The crash log says that the issue is within an audio thread `audio-io`. I attach the log file from the system for you to take a look. But basically here is the issue 

```
10  Foundation                    	       0x19e2b120c -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 188
11  AppKit                        	       0x1a091933c -[NSMenu itemArray] + 32
12  libqcocoa.dylib               	       0x107a7b078 0x107a5c000 + 127096
13  QtWidgets                     	       0x1075a83a8 QMenu::actionEvent(QActionEvent*) + 1216
14  QtWidgets                     	       0x10745940c QWidget::event(QEvent*) + 132
15  QtWidgets                     	       0x1075a6478 QMenu::event(QEvent*) + 1140
```

Hope you could help me out on this since this issue made my streams to fail and I need to recreate a new transmission so this could lead to potential viewers to think my lives do not behave well. 

I appreciate your attention and time to this matter! If you need further information from my end, please let me know. 
[OBS-move-transition-crash-report.txt](https://github.com/user-attachments/files/18031917/OBS-move-transition-crash-report.txt)

Mac description: 
Chip: Apple M3 Pro, 16-inch
MacOS: Sonoma 
Memory: 36GB

Move Plugin Version: 3.1.1

**Comment by felixolivares** [comment]
One more comment, I used the filter of Move Source, to move my video feeds from one scene to another trying to animate both cameras to go into scene and out of it as well. But during these transitions the crash happens.

**Comment by exeldro** [comment]
The audio thread destroys a source which causes qt to do things on the audio thread instead of the UI thread.
Move plugin can make this bug trigger faster, but is not the bug itself. Something is doing qt stuff not on the UI thread, so I would look at plugins that have UI elements, which the move plugin has not.

**Comment by felixolivares** [comment]
Thank you @exeldro for your notes. In deed, I actually removed all plugins and I still go the crash. I thought it was the Move plugin but it wasn't the one causing the crash. It seems that the issue was OBS itself. I tested the latest 31.0.0 RC1 version and it does not crash any more. So I believe this was a bug for the MacOS version. 

They have already released the 31.0.0 version to the public. I installed it and the crash is gone. Thank you for your help anyways!
